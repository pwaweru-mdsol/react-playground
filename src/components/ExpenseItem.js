import React from "react";
import { Link } from "react-router-dom";
import moment from 'moment';
import numeral from 'numeral';

export const ExpenseItem = ({ uuid, description, amount, createdAt }) => (
  <div>
    <Link to={`/edit/${uuid}`}>
      <h3>{description}</h3>
    </Link>
    <p>{numeral(amount / 100).format('$0,0.00')}</p>
    <p>{moment(createdAt).format('MMMM Do, YYYY')}</p>
  </div>
);

export default ExpenseItem;
