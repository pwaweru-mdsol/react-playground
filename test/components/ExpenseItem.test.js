import React from 'react';
import { shallow } from 'enzyme';
import ExpenseItem from '../../src/components/ExpenseItem';

const expense = {
    uuid: "1",
    description: "hello world !",
    amount: 45,      
    createdAt: '1'
}

test('should render ExpenseList component', () => {
    const wrapper = shallow(<ExpenseItem {...expense} />);
    expect(wrapper).toMatchSnapshot();
});