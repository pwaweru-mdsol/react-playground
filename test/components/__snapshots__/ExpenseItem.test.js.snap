// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpenseList component 1`] = `
<div>
  <Link
    replace={false}
    to="/edit/1"
  >
    <h3>
      hello world !
    </h3>
  </Link>
  <p>
    $0.45
  </p>
  <p>
    January 1st, 2001
  </p>
</div>
`;
